File: App/DAL/Interfaces/IRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace App.DAL.Interfaces
{
    interface IRepository<T> where T : class
    {
        IQueryable<T> GetAll();
        T Get(int id);
        void Update(T entity);
        void Create(T entity);
        void Delete(int id);
    }
}

File: App/Static/scripts/ts/classes/repos/IRepository.ts
import {ResponseResultSet} from "./ResponseResultSet";

export interface IRepository<TEntity>
{
	addAsync(entity: TEntity): Promise<ResponseResultSet>;
	
	update(entity: TEntity): Promise<ResponseResultSet>;
	
	delete(entity: TEntity): boolean;
	
	get(id: number): TEntity;
	
	getAll(): TEntity[];
}